### ISSUE TITLE
Waterfall: NPE in ServerHelper removeServer function

### ISSUE THREAD

**Issue by galexrt** [tavonkelly/BungeeServerManager#8]
Running Waterfall Build #388 when trying to edit an IP of a server I got an NPE:

```
[21:31:05 INFO]: Galexrt executed command: /svm edit mycoolserver-0 ip mycoolserver-0.mycoolserver:25565
[21:31:05 ERROR]: [/IP:42866|Galexrt] -> UpstreamBridge - encountered exception
java.lang.ExceptionInInitializerError: null
	at pw.teg.bsm.util.ServerHelper.removeServer(ServerHelper.java:40) ~[?:?]
	at pw.teg.bsm.commands.ServerManagerCommand.execute(ServerManagerCommand.java:288) ~[?:?]
	at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:197) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:148) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.connection.UpstreamBridge.handle(UpstreamBridge.java:158) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.protocol.packet.Chat.handle(Chat.java:81) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.netty.HandlerBoss.channelRead(HandlerBoss.java:105) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: org.yaml.snakeyaml.reader.ReaderException: special characters are not allowed
	at org.yaml.snakeyaml.reader.StreamReader.update(StreamReader.java:211) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.reader.StreamReader.ensureEnoughData(StreamReader.java:176) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.reader.StreamReader.ensureEnoughData(StreamReader.java:171) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.reader.StreamReader.peek(StreamReader.java:126) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.scanner.ScannerImpl.scanToNextToken(ScannerImpl.java:1177) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:287) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:227) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.parser.ParserImpl$ParseImplicitDocumentStart.produce(ParserImpl.java:195) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:158) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.parser.ParserImpl.checkEvent(ParserImpl.java:148) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:118) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:150) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:472) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at org.yaml.snakeyaml.Yaml.loadAs(Yaml.java:438) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.config.YamlConfiguration.load(YamlConfiguration.java:93) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at net.md_5.bungee.config.YamlConfiguration.load(YamlConfiguration.java:86) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:40370a3:388]
	at pw.teg.bsm.util.ConfigHelper.setupConfig(ConfigHelper.java:78) ~[?:?]
	at pw.teg.bsm.util.ConfigHelper.<clinit>(ConfigHelper.java:21) ~[?:?]
	... 41 more
[21:31:05 INFO]: [Galexrt] disconnected with: ExceptionInInitializerError : null @ pw.teg.bsm.util.ServerHelper:40
[21:31:05 INFO]: [/IP:42866|Galexrt] -> UpstreamBridge has disconnected
```

Please let me know if you need anymore information.

**Comment by tavonkelly** [comment]
Should be fixed in 068f179f29cbff3adcdf3d7c4fa99b0dc6591677

Can you download the latest version from Spigot (v1.1.1) and let me know if it's still an issue?

**Comment by galexrt** [comment]
@tavonkelly Now I am able to edit a server's IP again thanks!

Testing `/svm info SERVER_NAME` throws a NPE (the server's address is a DNS record if that can be a problem):
```
> svm info SERVER_NAME
[12:10:07 INFO]: CONSOLE executed command: /svm info SERVER_NAME
[12:10:07 INFO]: --- SERVER_NAME Info ---
[12:10:07 INFO]: Name: SERVER_NAME
[12:10:07 INFO]: An internal error occurred whilst executing this command, please check the console log for details.
[12:10:07 WARN]: Error in dispatching command
java.lang.NullPointerException: null
	at pw.teg.bsm.commands.ServerManagerCommand.execute(ServerManagerCommand.java:98) ~[?:?]
	at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:197) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:148) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at io.github.waterfallmc.waterfall.console.WaterfallConsole.runCommand(WaterfallConsole.java:30) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.minecrell.terminalconsole.SimpleTerminalConsole.processInput(SimpleTerminalConsole.java:87) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.minecrell.terminalconsole.SimpleTerminalConsole.readCommands(SimpleTerminalConsole.java:168) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.minecrell.terminalconsole.SimpleTerminalConsole.start(SimpleTerminalConsole.java:141) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.md_5.bungee.BungeeCordLauncher.main(BungeeCordLauncher.java:67) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
	at net.md_5.bungee.Bootstrap.main(Bootstrap.java:15) ~[waterfall.jar:git:Waterfall-Bootstrap:1.16-R0.4-SNAPSHOT:aa05f36:390]
```

**Comment by tavonkelly** [comment]
@galexrt this should be fixed in 507b67df7e1c18c337bf6b540a42485c33e5100b

Download version v1.1.2 from Spigot

**Comment by galexrt** [comment]
@tavonkelly Yep, that seems to have fixed it. Thanks for the quick help!
